staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -133,3 +136,165 @@ func TestWatchErrResultNotBlockAfterCancel(t *testing.T) {
 	cancel()
 	wg.Wait()
 }
+
+func TestWatchChanSync(t *testing.T) {
+	origCtx, store, _ := testSetup(t)
+	nativeKV := store.client.KV
+	initList, err := initStoreData(origCtx, store)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testCases := []struct {
+		name             string
+		watchKey         string
+		watcherMaxLimit  int64
+		addGetReactor    bool
```

Comment:
I don't see a need for it - why can't we just always add a reactor?

Reply:
Because the test cases use the same etcd server, if we add a reactor to each test case, the previous test case call will also add pod resources, causing the previous and subsequent test cases to affect each other. We need to perceive how many times the reactor is called in subsequent test cases

Reply:
> Because the test cases use the same etcd server,

We add a reactor client-side, so that doesn't matter...
I still don't see a need for it.